flink-streaming-java/src/main/java/org/apache/flink/streaming/api/transformations/TimestampsAndWatermarksTransformation.java

```diff
@@ -0,0 +1,102 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.transformations;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.common.eventtime.WatermarkStrategy;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.dag.Transformation;
+import org.apache.flink.streaming.api.operators.ChainingStrategy;
+import org.apache.flink.streaming.api.watermark.Watermark;
+
+import org.apache.flink.shaded.guava18.com.google.common.collect.Lists;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * A {@link PhysicalTransformation} for {@link Watermark}.
+ *
+ * @param <IN> The input and output type of the transformation.
+ */
+@Internal
+public class TimestampsAndWatermarksTransformation<IN> extends PhysicalTransformation<IN> {
+
+	private final Transformation<IN> input;
+	private final WatermarkStrategy<IN> watermarkStrategy;
+
+	private ChainingStrategy chainingStrategy = ChainingStrategy.DEFAULT_CHAINING_STRATEGY;
+
+	/**
+	 * Creates a new {@code Transformation} with the given name, output type and parallelism.
+	 *
+	 * @param name The name of the {@code Transformation}, this will be shown in Visualizations
+	 * 	and the Log
+	 * @param parallelism The parallelism of this {@code Transformation}
+	 * @param input The input transformation of this {@code Transformation}
+	 * @param watermarkStrategy The {@link WatermarkStrategy} to use
+	 */
+	public TimestampsAndWatermarksTransformation(
+		String name,
```

Comment:
Could you add an indentation here?

Reply:
@dawidwys sorry for the unmatched idea code style. I would like to modify the indentation.